src/txn-wal/src/txn_write.rs

```diff
@@ -285,7 +296,10 @@ where
                                     .commit_bytes
                                     .inc_by(u64::cast_from(batch.encoded_size_bytes()));
                             }
-                            handle.datas.put_write(data_write);
+                            let prev = handle
+                                .data_write_commit
+                                .insert(data_write.0.shard_id(), data_write);
+                            assert!(prev.is_none());
```

Comment:
note: I added an `assert_none!` macro recently to `ore` which debug prints the `Some(...)` value

Reply:
Done!